src/main/java/microsoft/exchange/webservices/data/property/complex/time/TimeZoneDefinition.java

```diff
@@ -114,13 +115,18 @@ public int compare(final TimeZoneTransition x, final TimeZoneTransition y) {
     if (x == y) {
       return 0;
     } else if (x != null && y != null) {
-      final AbsoluteDateTransition firstTransition = (AbsoluteDateTransition) x;
-      final AbsoluteDateTransition secondTransition = (AbsoluteDateTransition) y;
+      if (x instanceof AbsoluteDateTransition && y instanceof AbsoluteDateTransition) {
+        final AbsoluteDateTransition firstTransition = (AbsoluteDateTransition) x;
+        final AbsoluteDateTransition secondTransition = (AbsoluteDateTransition) y;
```

Comment:
Good point. It might make sense to implement compareTo() method for AbsoluteDateTransition, so it can be used in other places later. Plain TimeZoneTransition does not have anything to compare, so we do not really need compareTo() method.